### PR TITLE
core/infosync: prioritise p2p  protocols

### PR DESCRIPTION
**Pull request by corverroos**
Prioritise p2p protocols and make cluster-wise results available for querying by slot. This allows smooth upgrading to new protocols. Adds a `mock_alpha` protocol for ad-hoc testing in prod. Bumps prioritise to stable feature.

category: feature
ticket: #1421

### CODE REVIEW

File: core/infosync/infosync.go [hunk 1]
```diff
@@ -18,55 +18,135 @@ package infosync
...
+	return c
 }
 
 type Component struct {
 	prioritiser *priority.Component
 	versions    []string
+	protocols   []protocol.ID
+
+	mu      sync.Mutex
+	results []result
```
Comment (dB2510) [inline core/infosync/infosync.go:83]: do we need to trim this results?

**Review by dB2510 — APPROVED** [review]

File: core/infosync/infosync.go [hunk 2]
```diff
@@ -18,55 +18,135 @@ package infosync
...
-const topicVersion = "version"
+const (
+	topicVersion  = "version"
+	topicProtocol = "protocol"
+)
 
 // New returns a new infosync component.
-func New(prioritiser *priority.Component, versions []string) *Component {
+func New(prioritiser *priority.Component, versions []string, protocols []protocol.ID) *Component {
+	// Add a mock alpha protocol if alpha features enabled iot to test infosync in prod.
```
[the remark this answers is not in the document]
Comment (corverroos) [reply core/infosync/infosync.go:40]: https://www.abbreviations.com/abbreviation/in+order+to

File: core/infosync/infosync.go (hunk 1 above)
Comment (corverroos) [reply core/infosync/infosync.go:83]: don't think so, unless the config changes constantly which it shouldn't

File: core/infosync/infosync.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dB2510) [reply core/infosync/infosync.go:40]: use full form please 🙏

File: core/infosync/infosync.go (hunk 1 above)
Comment (corverroos) [reply core/infosync/infosync.go:83]: limiting to 100 results

**Pull request merged**